include/seqan3/range/view/take_line.hpp

```diff
@@ -0,0 +1,469 @@
+view_take_line(urng_t) -> view_take_line<std::remove_reference_t<urng_t>, require_eol>;
+
+// ============================================================================
+//  take_line_fn (adaptor definition)
+// ============================================================================
+
+/*!\brief View adaptor definition for view::take_line and view::take_line_or_throw.
+ * \tparam require_eol Whether to throw or not.
+ */
+template <bool require_eol>
+class take_line_fn : public pipable_adaptor_base<take_line_fn<require_eol>>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<take_line_fn<require_eol>>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       Call the view's constructor with the underlying view as argument.
+     * \returns     An instance of seqan3::detail::view_take_line.
+     */
+    template <view_concept urng_t>
+    static auto impl(urng_t urange)
+    {
+        return view_take_line<urng_t, require_eol>{std::move(urange)};
+    }
+
+    /*!\brief       Call the view's constructor with the underlying range wrapped in seqan3::view::all as argument.
+     * \returns     An instance of seqan3::detail::view_take_line.
+     */
+    template <viewable_range_concept urng_t>
+    static auto impl(urng_t && urange)
+    {
+        return impl(view::all(std::forward<urng_t>(urange)));
+    }
+};
+
+} // namespace seqan3::detail
+
+// ============================================================================
+//  view::take_line (adaptor instance definition)
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief               A view adaptor that returns a single line from the underlying range or the full range if there
+ *                      is no newline.
+ * \tparam urng_t       The type of the range being processed. See below for requirements. [template parameter is
+ *                      omitted in pipe notation]
+ * \param[in] urange    The range being processed. [parameter is omitted in pipe notation]
+ * \returns             All characters of the underlying range up until, but excluding a unix or windows end-line
+ *                      (`\n` or `\r\n`). See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This adaptor returns a single line **excluding** the end-line character(s), *but moving the cursor behind them
+ * for single-pass ranges.* I.e. for all ranges that satisfy seqan3::forward_range_concept this is the same as calling
+ * ```cpp
+ * ranges::view::take_while([] (auto const & l) { return (l != '\r') && (l != '\n'); })
+ * ```
+ * but for ranges that are single pass this means that the endline is also consumed.
```

Comment:
_single pass ranges_ might be easier to understand here instead of _ranges that are single pass_